### PR TITLE
add another python script to generate google id token.

### PR DESCRIPTION
**Pull request by qiwzhang**

### CODE REVIEW

File: auth/generate-google-id-jwt2.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+signer = crypt.RSASigner.from_service_account_info(service_account_info)
+
+now = int(time.time())
+payload = {
+    'iat': now,
+    # expires after one hour.
+    'exp': now + 3600,
+    'iss': service_account_info['client_email'],
+    # the URL of the target service.
+    'target_audience': 'https://registry.endpoints.robco-166608.cloud.goog',
```
Comment (liminw) [inline auth/generate-google-id-jwt2.py:27]: This is the target service name. Do you want to hardcode it as robco service?

File: auth/generate-google-id-jwt2.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    service_account_info = json.load(fh)
+
+signer = crypt.RSASigner.from_service_account_info(service_account_info)
+
+now = int(time.time())
+payload = {
+    'iat': now,
+    # expires after one hour.
+    'exp': now + 3600,
+    'iss': service_account_info['client_email'],
```
Comment (liminw) [inline auth/generate-google-id-jwt2.py:25]: We can also have a 'sub' claim, which is the same as 'iss". But it is optional.

File: auth/generate-google-id-jwt2.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+from pprint import pprint
+
+from google.auth import crypt, jwt
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+# generates a signed JSON Web Token
+
+with open('/usr/local/google/home/ensonic/robco_gob/cloud-robotics/src/bootstrap/robot/robot-credentials.json', 'r') as fh:
```
Comment (liminw) [inline auth/generate-google-id-jwt2.py:15]: do you want to hardcode the file path?

File: auth/generate-google-id-jwt2.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+headers = {"Content-Type": "application/x-www-form-urlencoded"}
+response = requests.post('https://www.googleapis.com/oauth2/v4/token', data=params, headers=headers)
+res = response.json()
+
+pprint(res)
+
+# request
+
+headers = {'Authorization': 'Bearer {}'.format(res['id_token'])}
+response = requests.get('https://registry.ensonic.cloudrobotics.com/v1/testauth', headers=headers, verify=False)
```
Comment (liminw) [inline auth/generate-google-id-jwt2.py:51]: This is robco's service name and path. do you want to hardcode it here?

Comment (qiwzhang) [reply auth/generate-google-id-jwt2.py:51]: I intended to use this script as a reference.  Users need to modify it with their data when they use it.

File: auth/generate-google-id-jwt2.py (hunk 3 above)
Comment (ensonic) [reply auth/generate-google-id-jwt2.py:15]: Please use os.environ['GOOGLE_APPLICATION_CREDENTIALS'] or a dummy path like '/path/to/service_account.json'

File: auth/generate-google-id-jwt2.py (hunk 1 above)
Comment (ensonic) [reply auth/generate-google-id-jwt2.py:27]: Please don't.

File: auth/generate-google-id-jwt2.py (hunk 4 above)
Comment (ensonic) [reply auth/generate-google-id-jwt2.py:51]: Please also change to e.g. the bookstore example.

File: auth/generate-google-id-jwt2.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+now = int(time.time())
+payload = {
+    'iat': now,
+    # expires after one hour.
+    'exp': now + 3600,
+    'iss': service_account_info['client_email'],
+    # the URL of the target service.
+    'target_audience': 'https://registry.endpoints.robco-166608.cloud.goog',
+    # Google token endpoints URL, FIXME: use discovery doc
+    'aud': 'https://www.googleapis.com/oauth2/v4/token'
```
Comment (ensonic) [inline auth/generate-google-id-jwt2.py:29]: I don't think this URL should be hardcoded either (se also below). This should probably taken from api discovery.

**Comment by qiwzhang** [comment]
Move this file to a different place: 

https://github.com/GoogleCloudPlatform/python-docs-samples/pull/1112

**Pull request closed without merging**